### ISSUE TITLE
Utilizing a hosts file

### ISSUE THREAD

**Issue by pnelsonsr** [sameersbn/docker-gitlab#307]
I had been able to pull in the docker servers hosts file into the container but this is no longer working.  It took me awhile to figure out what was happening when startup just failed after I upgraded to 7.10.1 and I finally replace the `-d` with `-it --rm` and then I saw that it was failing to connect to the db.  I then changed the `-e "DB_HOST=real ip address"` and then gitlab started.  I then exec'ed with 
`docker exec -it gitlab bash` and then looked at the `/etc/hosts` file and it was not replaced with the docker servers hosts file.  The command that I had been running was `-v /root/dev/conf/hosts/hosts-all:/etc/hosts:ro` but its just not replacing the gitlab `/etc/hosts` file.  Any suggestions?  I am able to work around it with specific the specific IP but this is problematic as our IPs change.

**Comment by sameersbn** [comment]
@pnelsonsr `docker run -it -v /etc/hosts:/etc/hosts:ro --rm sameersbn/ubuntu:latest cat /etc/hosts` does not work for me as well. This is not recommended practice. I suggest you setup a local DNS server to setup a local domain. Refer to my blog post [Deploying a DNS Server using Docker](http://www.damagehead.com/blog/2015/04/28/deploying-a-dns-server-using-docker/) for some pointers.

**Comment by netflash** [comment]
Try to use this option

```
man docker run

--add-host=[]
          Add a custom host-to-IP mapping (host:ip)

       Add a line to /etc/hosts. The format is hostname:ip.  The --add-host option can be set multiple times.
```

**Comment by sameersbn** [comment]
@pnelsonsr that's a good suggestion by @netflash, should fix your issue.

**Comment by pnelsonsr** [comment]
@netflash thanks for the command line option.  The problem is that it is static.  Our hosts-all file on our all our docker servers are fequently updated and because we could map it into docker all docker containers were updated at the time hosts-all was.  My temporary solution was to get the ip (within the docker run command) with python -c "import socket; print socket.gethostbyname(\"hostname here\")", but again this is to static.

Utimately the real solution that I've been putting off (because I could with -v) is a DNS server and @sameersbn has a great docker container for that!  We are working on that now, @sameersbn your docs rule!

**Comment by pnelsonsr** [comment]
So I implemented your docker bind container and it is working.  I removed the 
-v /root/dev/conf/hosts/hosts-all:/etc/hosts:ro
and added
--dns='IP of the DNS container'
and this works.  But...
I have to pull the IP of the actual container to get gitlab to recognize the DNS.  I think this may be because I don't include the network range that includes the docker gitlab IP.  When I tried putting the host docker server IP in 
--dns='IP of the docker server'
this did not work.  I'm going to try putting the docker network in the ACL of bind and see it that works.  Is that how you do it?
